opentech/apply/determinations/views.py

```diff
@@ -240,27 +241,37 @@ def get_success_url(self):
     def form_valid(self, form):
         super().form_valid(form)
 
-        if not self.object.is_draft:
-            messenger(
-                MESSAGES.DETERMINATION_OUTCOME,
-                request=self.request,
-                user=self.object.author,
-                submission=self.object.submission,
-                related=self.object,
+        if self.object.is_draft:
+            return HttpResponseRedirect(self.submission.get_absolute_url())
+
+        messenger(
+            MESSAGES.DETERMINATION_OUTCOME,
+            request=self.request,
+            user=self.object.author,
+            submission=self.object.submission,
+            related=self.object,
+        )
+        transition = transition_from_outcome(form.cleaned_data.get('outcome'), self.submission)
+
+        if self.object.outcome == NEEDS_MORE_INFO:
+            # We keep a record of the message sent to the user in the comment
+            Activity.comments.create(
+                message=self.object.stripped_message,
+                timestamp=timezone.now(),
+                user=self.request.user,
+                submission=self.submission,
+                related_object=self.object,
             )
-            transition = transition_from_outcome(form.cleaned_data.get('outcome'), self.submission)
-
-            if self.object.outcome == NEEDS_MORE_INFO:
-                # We keep a record of the message sent to the user in the comment
-                Activity.comments.create(
-                    message=self.object.stripped_message,
-                    timestamp=timezone.now(),
-                    user=self.request.user,
-                    submission=self.submission,
-                    related_object=self.object,
-                )
 
-            self.submission.perform_transition(transition, self.request.user, request=self.request, notify=False)
+        self.submission.perform_transition(
+            transition,
+            self.request.user,
+            request=self.request,
+            notify=False,
+        )
+
+        if self.object.outcome == ACCEPTED:
```

Comment:
I've realised I missed an acceptance criteria. This should only happen when Accepted and there isnt another stage in the application workflow.

I don't know if we have a method to check that yet, you might want to do some digging. But pseudo code:
```python
 submission.workflow.stages.index(submission.stage) == len(submission.workflow.stages)
```
/musing
Maybe something to add to the workflow class

Reply:
Fixed in 8baa461f